src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_StackPanel.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.UI.Xaml.Controls;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Private.Infrastructure;
+
+namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml_Controls
+{
+	[TestClass]
+	public class Given_StackPanel
+	{
+		[TestMethod]
+		[RunsOnUIThread]
+		public async Task When_InsertingChildren_Then_ResultIsInRightOrder()
+		{
+			// This is an illustration of the bug https://github.com/unoplatform/uno/issues/3543
+
+			var pnl = new StackPanel();
+			pnl.Children.Add(new Button { Content = "abc" });
+			pnl.Children.Insert(0, new TextBlock { Text = "TextBlock" });
+			pnl.Children.Insert(0, new TextBox());
+
+			TestServices.WindowHelper.WindowContent = pnl;
+
+			using var _ = new AssertionScope();
```

Comment:
Codacy found an issue: [Remove the unused local variable '_'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6204376)